org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenExecutionContext.java

```diff
@@ -366,15 +366,19 @@ private <V> V executeBare(MavenProject project, ICallable<V> callable, IProgress
       }
       if(project != null) {
         mavenSession.setCurrentProject(project);
-        mavenSession.setProjects(Collections.singletonList(project));
+        List<MavenProject> projects = Collections.singletonList(project);
+        mavenSession.setProjects(projects);
+        mavenSession.setAllProjects(projects);
       }
       return callable.call(this, IProgressMonitor.nullSafe(monitor));
     } finally {
       Thread.currentThread().setContextClassLoader(origTCCL);
       repositorySession.setTransferListener(origTransferListener);
       if(project != null) {
         mavenSession.setCurrentProject(origProject);
-        mavenSession.setProjects(origProjects != null ? origProjects : Collections.<MavenProject> emptyList());
+        List<MavenProject> projects = origProjects != null ? origProjects : Collections.<MavenProject> emptyList();
+        mavenSession.setProjects(projects);
```

Comment:
"projects" and "all projects" are not always the same!
So if you want to reset it both values must be saved before hands.

In general for plugins calling getAllProjects is bad, the method is marked with:
`/** This is a provisional method and may be removed */`
and it should actually only be used in the setup phase.

Reply:
> "projects" and "all projects" are not always the same!
So if you want to reset it both values must be saved before hands.

Fixed.

> In general for plugins calling getAllProjects is bad, the method is marked with:
/** This is a provisional method and may be removed */
and it should actually only be used in the setup phase.

maven-quarkus-plugin works from the mvn command line. See https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/DefaultMaven.java#L656

Reply:
"works" does not mean it is good style, I don't know why `maven-quarkus-plugin` is using `getAllProjects`, but the value is not the same as `getProjects` and should only be used in very rare cases (probably never).

`getProject` return all projects that are selected for a reactor build and `getAllProjects` will contain all possible projects that are discovered through modules, so a plugin using `getAllProjects` will access projects that are probably never build.